### PR TITLE
[SPARK-9319][SPARKR] Add support for setting column names, types

### PR DESCRIPTION
**Pull request by felixcheung**
Add support for for colnames, colnames<-, coltypes<-
Also added tests for names, names<- which have no test previously.

I merged with PR 8984 (coltypes). Clicked the wrong thing, crewed up the PR. Recreated it here. Was #9218 

@shivaram @sun-rui

### CODE REVIEW

**Comment by felixcheung** [comment]
any more comment?

File: R/pkg/R/types.R [hunk 1]
```diff
@@ -41,3 +41,11 @@ COMPLEX_TYPES <- list(
 
 # The full list of data types.
 DATA_TYPES <- as.environment(c(as.list(PRIMITIVE_TYPES), COMPLEX_TYPES))
+
+# An environment for mapping R to Scala, names are R types and values are Scala types.
+rToScalaTypes <- new.env()
```
Comment (sun-rui) [inline R/pkg/R/types.R]: better name: rToSQLTypes

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -293,6 +294,121 @@ setMethod("names<-",
...
+            rTypes[naIndices] <- types[naIndices]
+
+            rTypes
+          })
+
+#' coltypes
+#'
+#' Set the column types of a DataFrame.
+#'
+#' @name coltypes<-
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: formalize the description

File: R/pkg/R/DataFrame.R [hunk 3]
```diff
@@ -293,6 +294,121 @@ setMethod("names<-",
...
+          function(x, value) {
+            sdf <- callJMethod(x@sdf, "toDF", as.list(value))
+            dataFrame(sdf)
+          })
+
+#' coltypes
+#'
+#' Get column types of a DataFrame
+#'
+#' @name coltypes
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: formalize the description

File: R/pkg/R/DataFrame.R [hunk 4]
```diff
@@ -293,6 +294,121 @@ setMethod("names<-",
             }
           })
 
+#' @rdname columns
+#' @name colnames
+setMethod("colnames",
+          signature(x = "DataFrame"),
+          function(x, do.NULL = TRUE, prefix = "col") {
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: Is it possible to remove default value for do.NULL and prefix, and throws unsupported message when they are not missing?

File: R/pkg/R/DataFrame.R (hunk 2 above)
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: Could you clarify?

Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: I mean make the style consistent with others. for example,
@param x (DataFrame)  -> @param x A DataFrame.

**Comment by felixcheung** [comment]
updated from comments.

**Comment by felixcheung** [comment]
@sun-rui any more comment?

**Comment by felixcheung** [comment]
@sun-rui ?

**Comment by sun-rui** [comment]
Will take a look soon

**Comment by felixcheung** [comment]
Could we please review this? I think it would be great to get matching coltype<- in Spark 1.6

File: R/pkg/R/DataFrame.R [hunk 5]
```diff
@@ -290,6 +292,121 @@ setMethod("names<-",
...
+#' @name colnames
+setMethod("colnames",
+          signature(x = "DataFrame"),
+          function(x) {
+            columns(x)
+          })
+
+#' @rdname columns
+#' @name colnames<-
+setMethod("colnames<-",
```
Comment (shivaram) [inline R/pkg/R/DataFrame.R]: `colnames<-` also needs to be added to the exported list of functions ?

File: R/pkg/R/types.R [hunk 6]
```diff
@@ -41,3 +41,11 @@ COMPLEX_TYPES <- list(
 
 # The full list of data types.
 DATA_TYPES <- as.environment(c(as.list(PRIMITIVE_TYPES), COMPLEX_TYPES))
+
+# An environment for mapping R to Scala, names are R types and values are Scala types.
+rToSQLTypes <- new.env()
```
Comment (shivaram) [inline R/pkg/R/types.R]: Can we follow the same style as above ? `as.environment(list())`

**Comment by shivaram** [comment]
Couple of minor comments, but otherwise change looks good.

**Comment by shivaram** [comment]
LGTM. Thanks @felixcheung

**Pull request closed without merging**